hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -281,12 +331,40 @@ private boolean checkKeyReclaimable(
         return false;
```

Comment:
This means we won't reclaim those blocks (e.g. added by Hsync) in deletedTable? while actually the blocks **could** be deleted if the block ID doesn't match.

This may deserve a TODO here (Add block ID checking). Also, add more comments regarding the desired GC behavior.

Make sure when deferring this entry to the next snapshot or active DB, it is merged properly into the same OmKeyInfo. The blocks cannot be thrown away without reclamation (leaking blocks on SCMs/DNs otherwise).

Reply:
We will reclaim these blocks on the `deletedTable`, here these uncommitted blocks are put in `reclaimList` which will be reclaimed by `KeyDeletingService`.